app/services/embedded_content_finder_service.rb

```diff
@@ -8,20 +8,20 @@ class EmbeddedContentFinderService
   def fetch_linked_content_ids(details, locale)
     content_references = details.values.map { |value|
       find_content_references(value)
-    }.flatten.compact.uniq
+    }.flatten.compact
 
-    check_all_references_exist(content_references, locale)
+    check_all_references_exist(content_references.uniq, locale)
     content_references.map(&:content_id)
   end
 
   def find_content_references(value)
     case value
     when Array
-      value.map { |item| find_content_references(item) }.flatten
+      value.map { |item| find_content_references(item) }.flatten.uniq
     when Hash
-      value.map { |_, v| find_content_references(v) }.flatten
+      value.map { |_, v| find_content_references(v) }.flatten.uniq
     when String
-      value.scan(EMBED_REGEX).map { |match| ContentReference.new(document_type: match[1], content_id: match[2], embed_code: match[0]) }.uniq
+      value.scan(EMBED_REGEX).map { |match| ContentReference.new(document_type: match[1], content_id: match[2], embed_code: match[0]) }
```

Comment:
I guess there might not be an easier way of getting the count, and maybe not an issue now, but will having duplicate Links mean adding more links to iterate through when dependencies get updated? so if there are 10s of blocks in a host document, that could trigger 100s more calls and jobs throughout the host documents lifetime?

Reply:
Good call, but it looks like DependencyResolution will return an array of `content_id`s, which will then get uniqued out when we query (see https://github.com/alphagov/publishing-api/blob/4a679f153c286c248aa3f4fbf81b0a67278cede7/app/sidekiq/dependency_resolution_job.rb#L4) so I _think_ we're fine